neo/SmartContract/InteropService.cs

```diff
@@ -236,14 +236,16 @@ private static bool Runtime_Serialize(ApplicationEngine engine)
 
         private static bool Runtime_GetNotifications(ApplicationEngine engine)
         {
-            var data = engine.CurrentContext.EvaluationStack.Pop().GetByteArray();
-            if (data.Length != UInt160.Length) return false;
+            byte[] data = engine.CurrentContext.EvaluationStack.Pop().GetByteArray();
+            if ((data.Length != 0) && (data.Length != UInt160.Length)) return false;
             if (!engine.CheckArraySize(engine.Notifications.Count)) return false;
 
-            var hash = new UInt160(data);
             IEnumerable<NotifyEventArgs> notifications = engine.Notifications;
-            if (!hash.Equals(UInt160.Zero))
+            if (data.Length == UInt160.Length) // must filter by scriptHash
+            {
+                var hash = new UInt160(data);
```

Comment:
we can check if is 0 here too, or is not needed?

Reply:
Not needed... it only passes if it's 0 or 20... Originally, it takes all Notifications (which is 0 behavior). So, if it's 20, then it filters (otherwise, if it's zero, just keep all notifications).
Any size different than zero or 20, it faults.

Question is: can we create a `null` stackitem? I don't now.. and even if we do that, is it better than byte[0] here? I prefer handling constructed stuff than null stuff.
On the other hand, if we put on devpack a ScriptHash type (or UInt160), the option there will be that or null... must see how this is interpreted on nvm.

That's why I prefer this way.